cosmo/__main__.py

```diff
@@ -60,6 +60,9 @@ def noop(*args, **kwargs):
     for l2vpn in cosmo_data["l2vpn_list"]:
         if not l2vpn["name"].startswith("WAN: "):
             continue
+        if l2vpn['type'] == "VPWS" and len(l2vpn['terminations']) != 2:
```

Comment:
You also want to check, that VPWS L2VPNs only contain interfaces?

Reply:
Using VLANs to expand to interfaces is totally valid, even for VPWS setups.
But I get that we can't guarantee that we only have two endpoints then. What do you think? Shall we expect explicit interface configuration for VPWS tunnels?

Reply:
Netbox only allows two terminations for VPWS, but I am not sure, if those two terminations also can be two different VLANs. 

Therefore, I would postulate, that if we want to connect two interfaces, we should use two interface terminations and and if we want to connect two VLANs, we should use VLAN interfaces - but I am not sure, if this is a thing in VPWS.

Reply:
> Netbox only allows two terminations for VPWS

this is gonna be fun when we implement Multi-Homed Ethernet Segments for VPWS. But that's our problem for later.

Anyways, VPWS circuits on VLANs are a thing. But not to confuse two things:
What we are doing here is not connecting a VPWS circuit to a VLAN bridge domain, but expanding VLANs to Interfaces, to connect a circuit to it. (Even though in an EVPN (non-VPWS) circuit we implicitly configure a bridge domain by doing so, so here that might make sense)

TL;DR: All this can be a bit confusing. So I'd follow your suggestion and only allow interface terminations on VPWS circuits - as it's also possible to terminate a VPWS circuit on a VLAN sub-interface.